crates/dojo-world/src/manifest_test.rs

```diff
@@ -79,3 +83,32 @@ async fn test_manifest_loads_empty_world_from_mock_remote() {
         }
     )
 }
+
+#[tokio::test]
+async fn test_migration_from_remote() {
+    let target_dir = Utf8PathBuf::from_path_buf("../../examples/ecs/target/dev".into()).unwrap();
+
+    let sequencer = Sequencer::start().await;
+    let account = sequencer.account();
+    let env_config = EnvironmentConfig {
+        rpc: Some(sequencer.url()),
+        account_address: Some(account.address),
+        private_key: Some(account.private_key),
+        ..EnvironmentConfig::default()
+    };
+    let world = World::from_path(target_dir.clone(), WorldConfig::default(), env_config.clone())
+        .await
+        .unwrap();
+
+    let mut migration = world.prepare_for_migration(target_dir.clone()).await.unwrap();
+    migration.execute().await.unwrap();
+
+    let local_manifest = Manifest::load_from_path(target_dir.join("manifest.json")).unwrap();
+
+    let remote_manifest =
+        Manifest::from_remote(local_manifest.world.unwrap(), env_config.provider().unwrap(), None)
```

Comment:
manifest only stores the class hashes tho. here you are passing the local world class hash instead of the contract address `from_remote`